src/taco-remote/lib/buildManager.ts

```diff
@@ -431,6 +432,17 @@ class BuildManager {
                 } else {
                     self.buildMetrics.failed++;
                 }
+                
+                utils.TelemetryHelper.generate("build",
+                    (telemetry: utils.TelemetryGenerator) => {
```

Comment:
Do we care about getting telemetry for the cases where this method returns early? or throws an exception, or crashes?
A best practice is to generate telemetry as soon as you get the data, so in case that the code follows a different path, or has an issue, you'll still get it.
If we aren't interested in those scenarios, then we can leave it as is. If we want to capture them, we should refactor the telemetry so it's generated as soon as possible.